Integrations/Integration-Wildfire.yml

```diff
@@ -0,0 +1,261 @@
+        context.File = [];
+        var ep = {};
+
+        ep.Type = body.filetype;
+        ep.MD5 = body.md5;
+        ep.SHA1 = body.sha1;
+        ep.SHA256 = body.sha256;
+        ep.Size = body.size;
+        if(body.filename){
+            ep.Name = body.filename;
+        }
+
+        context.File.push(ep);
+
+        if(body.malware == 'yes'){
+            context['File/Malicious'] = [];
+            context['File/Malicious'].push(ep);
+        }
+        return context;
+    };
+
+
+    // The command input arg holds the command sent from the user.
+    switch (command) {
+      // This is the call made when pressing the integration test button.
+      case 'test-module':
+          var res = http(
+              url,
+              {
+                  Method: 'POST',
+                  Body: 'apikey=37ddffe5490c01835b6645fb6ecfe9c9&format=xml&hash=7f638f13d0797ef9b1a393808dc93b94',
+                  Headers: headers
+              }
+          );
+
+          if(res.StatusCode==200)
+              return 'ok';
+          else
+              return 'Something went wrong';
+          break;
+
+      case 'WildFireReport':
+        var url = params.server + urlDict.report;
+        var hash = args.md5 || args.sha256;
+        if(!hash)
+            throw 'Missing md5/sha256';
+
+        var bodyXML = 'apikey='+params.token+'&format=xml&hash='+hash;
+        var resXML = sendRequest(url, bodyXML, headers);
+        var result = JSON.parse(x2j(resXML));
+        var report = result.wildfire.task_info.report;
+        var file_info = result.wildfire.file_info;
+
+        var context = parseFile(result.wildfire.file_info);
+
+        if(args.format=='pdf'){
+          var bodyPDF = 'apikey='+params.token+'&format=pdf&hash='+hash;
+          var resPDF = sendRequest(url, bodyPDF, headers);
+
+          var currentTime = new Date();
+          var fileName = command + '_at_' + currentTime.getTime();
+
+          return {Type: entryTypes.note,
+                  FileID: saveFile(resPDF),
+                  File: fileName,
+                  Contents: fileName,
+                  EntryContext: context
+          };
+        }
+        else{ /* default format is xml*/
+          var md = resToMD(file_info,'File info');
+          for (var i=0; i<report.length; i++){
+              md += resToMD(report[i],'Report '+i);
+          }
+          return {Type: entryTypes.note,
+                  Contents: result,
+                  ContentsFormat: formats.json,
+                  HumanReadable: md, /*todo*/
+                  ReadableContentsFormat: formats.markdown,
+                  EntryContext: context
+          };
+        }
+        break;
+
+      case 'WildFireUpload':
+          var url = params.server + urlDict.upload;
+          var result = sendMultipartRequest(url, headers, args.upload, params.token);
+          var fileinfo = result.wildfire['upload-file-info'];
+          var context = parseFile(fileinfo);
```

Comment:
About this one - I needed the fileinfo for lines 227 and 228. I figured creating a var would make it cleaner. Is it wrong?